cmd/simulator/load/loader.go

```diff
@@ -129,5 +138,12 @@ func ExecuteLoader(ctx context.Context, config config.Config) error {
 		return err
 	}
 	log.Info("Tx agents completed successfully.")
+
+	if config.ExposeMetrics {
+		// Start a prometheus server to expose individual tx metrics
+		http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{Registry: reg}))
+		log.Info("localhost:8082/metrics")
+		http.ListenAndServe(":8082", nil)
```

Comment:
We should handle the error returned here and make sure this handles ctrl c correctly.

Can we update this so that the server runs at the same time as the test so we can ingest metrics while it's running?

Blocking forever to run a metrics server is not great behavior here. Would it make sense to optionally hit this HTTPServer ourselves to get the result rather than running the server and blocking?

Reply:
fixed